lib/centurion/service.rb

```diff
@@ -63,10 +63,11 @@ def image=(image)
       @image = image
     end
 
-    def build_config(server_hostname, &block)
+    def build_config(server_hostname, hostname = nil)
+
       container_config = {}.tap do |c|
         c['Image'] = image
-        c['Hostname'] = yield server_hostname if block_given?
+        c['Hostname'] = hostname if hostname
```

Comment:
As of 1.7.0 the hostname of the container is not overridden if no hostname was provided as an override. Or I misunderstood [the test](https://github.com/newrelic/centurion/blob/master/spec/service_spec.rb#L97).